### PR TITLE
FS-1938: tweak key application data report

### PR DESCRIPTION
**Pull request by gidsg**
* Changes to report requested by product manager (change report fields, and filter to only Submitted applications)
* Changes to allow report to succeed even if no matches found by supplying headers (previously it would throw an exception in that case
* Added test

### CODE REVIEW

File: db/models/aggregate_functions.py [hunk 1]
```diff
@@ -272,63 +286,60 @@ def get_report_for_all_applications(application_id=None):
...
-            "asset_type": None,
-            "capital": None,
-            "geography": None,
-            "organisation_type": None,
-            "revenue": None,
-        }
+        return_json = {field: None for field in KEY_REPORT_FIELDS}
         stored_forms = [form.as_json() for form in application.forms]
-        list_of_forms = [
+        report_fields = [
```
Comment (Adam-W1) [inline db/models/aggregate_functions.py:294]: Im just wondering the diff between KEY_REPORT_FIELDS and this List, is the reason they are separate that they may diverge at some point?

Comment (gidsg) [reply db/models/aggregate_functions.py:294]: Yes, one is the headers for the report and this is the field mapping. I think we could link them but we'd need to extract the mapping out so we can derive the headers. I'll have a go at that.

Comment (gidsg) [reply db/models/aggregate_functions.py]: @Adam-W1 OK I've made that change in [9c301d5](https://github.com/communitiesuk/funding-service-design-application-store/pull/96/commits/9c301d5f1357fe647974c831b6fbfb2ff4470f40) if you can re-review.

**Review by Adam-W1 — APPROVED** [review]
Nice and intuitive 👌

**Pull request merged**